Sources/Foundation/NSData.swift

```diff
@@ -670,7 +671,7 @@ open class NSData : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
      - parameter options:    Options for handling invalid input
      - returns:              The decoded bytes.
      */
-    private static func base64DecodeBytes(_ bytes: [UInt8], options: Base64DecodingOptions = []) -> [UInt8]? {
+    private static func base64DecodeBytes(_ bytes: UnsafeRawBufferPointer, options: Base64DecodingOptions = []) -> [UInt8]? {
```

Comment:
Rather than accepting an `UnsafeRawBufferPointer` it would be safer to change the signature to :

`private static func base64DecodeBytes<T: Collection>(_ bytes: T, options: Base64DecodingOptions = []) -> [UInt8]? where T.Element == UInt8 {`

Then it could be called using:
```
NSData.base64DecodeBytes(base64String.utf8, ...)

and

NSData.base64DecodeBytes(base64Data, ...)
```

This has a 2nd benefit of avoiding the use of `.utf8CString` which may have to reallocate if it needs more space since it appends a `NUL` to the end of the buffer. This also eliminates the use of `withUnsafeBytes` in the 2nd initialiser.

What do you think?